pkg/agent/controller/egress/egress_controller.go

```diff
@@ -585,10 +563,34 @@ func (c *EgressController) syncEgress(egressName string) error {
 		}
 		exist = false
 	}
+	// Do not proceed if EgressIP is empty.
+	if egress.Spec.EgressIP == "" {
+		return nil
+	}
```

Comment:
I have a problem here. When a user updates egress.Spec.EgressIP from an assigned IP to "". How does the agent unassign the EgressIP?

Reply:
It's handled in `uninstallEgress`, for both cases: egress is removed, egressIP is updated, regardless of it's updated to "" or another IP.

Reply:
I got it, that´s great, Thanks.
```
	// If the EgressIP changes, uninstalls this Egress first.
	if exist && eState.egressIP != egress.Spec.EgressIP {
		if err := c.uninstallEgress(egressName, eState); err != nil {
			return err
		}
		exist = false
	}
```